src/main/java/com/mongodb/FongoDBCollection.java

```diff
@@ -827,7 +852,9 @@ public static DBObject applyProjections(DBObject result, DBObject projectionObje
       } else if (ExpressionParser.isDbObject(projectionValue)) {
         project = true;
         projectionFields.add(projectionKey);
-      } else if (!projectionValue.toString().equals("text")) {
+      } else if (projectionValue.toString().equals("text")) {
+        included = true;
```

Comment:
This fixes the earlier CI failure. Looks like it's a hidden bug I uncovered with this change.